src/node_group.cpp

```diff
@@ -45,16 +67,12 @@ std::array<float, 3> NodeGroup::sop() const {
     return { center_x, center_y, center_z };
 }
 
-const std::set<const Tile*, TilePCompare> NodeGroup::get_tiles() const{
+const std::set<const Tile*, TilePCompare> NodeGroup::get_tiles() const {
     return std::set<const Tile*, TilePCompare>(tiles.begin(), tiles.end());
 }
 
-std::set<Tile*, TilePCompare> NodeGroup::get_tiles(){
-    return tiles;
-}
-
-inline bool NodeGroup::adjacent_to(const NodeGroup* other) const{
-    return (std::find(adjacent.begin(), adjacent.end(), other) == adjacent.end());
+inline bool NodeGroup::adjacent_to(NodeGroup* other) const{
+    return (adjacent.find(other)) == adjacent.end();
```

Comment:
Inline functions must be defined in header files, otherwise their source is not visible to compilation units (cpp files) and thus they are not inlined